pkg/extensionAPI/errorHandlingExtension_test.go

```diff
@@ -206,6 +206,78 @@ func (suite *ErrorHandlingExtensionSuite) TestUpgradeUnsupported() {
 	suite.Nil(instance)
 }
 
+// SupportsListInstances
+
+func (suite *ErrorHandlingExtensionSuite) TestSupportsListInstancesUnsupportedMissingMethod() {
+	suite.rawExtension.FindInstances = nil
+	suite.False(suite.extension.SupportsListInstances(createMockContext(), "version"))
+}
+
+func (suite *ErrorHandlingExtensionSuite) TestSupportsListInstancesSupportedReturnsResult() {
+	suite.rawExtension.FindInstances = func(context *context.ExtensionContext, version string) []*JsExtInstance {
+		return []*JsExtInstance{}
+	}
+	suite.True(suite.extension.SupportsListInstances(createMockContext(), "version"))
+}
+
+func (suite *ErrorHandlingExtensionSuite) TestSupportsListInstancesSupportedReturnsGeneralError() {
+	suite.rawExtension.FindInstances = func(context *context.ExtensionContext, version string) []*JsExtInstance {
+		panic(mockErrorMessage)
+	}
+	suite.True(suite.extension.SupportsListInstances(createMockContext(), "version"))
+}
+
+// ListInstances
+
+func (suite *ErrorHandlingExtensionSuite) TestListInstancesSuccessful() {
+	suite.rawExtension.FindInstances = func(context *context.ExtensionContext, version string) []*JsExtInstance {
+		return []*JsExtInstance{{Id: "inst1", Name: "Inst 1"}, {Id: "inst2", Name: "Inst 2"}}
+	}
+	instances, err := suite.extension.ListInstances(createMockContext(), "version")
+	suite.NoError(err)
+	suite.Equal([]*JsExtInstance{{Id: "inst1", Name: "Inst 1"}, {Id: "inst2", Name: "Inst 2"}}, instances)
+}
+
+func (suite *ErrorHandlingExtensionSuite) TestListInstancesFails() {
+	suite.rawExtension.FindInstances = func(context *context.ExtensionContext, version string) []*JsExtInstance {
+		panic(mockErrorMessage)
+	}
+}
```

Comment:
Shouldn't you exercise the `ListInstances` method here. Looks incomplete to me.

Reply:
Ooops...